components/places/src/storage/history.rs

```diff
@@ -1425,6 +1425,27 @@ pub fn get_visit_count(db: &PlacesDb, exclude_types: VisitTransitionSet) -> Resu
     Ok(count)
 }
 
+pub fn get_visit_count_for_host_before_timestamp(
+    db: &PlacesDb,
+    host: &str,
+    before: Timestamp,
+) -> Result<i64> {
+    let count = db.query_row_and_then_cachable(
+        "SELECT COUNT(*)
+        FROM moz_historyvisits
+        JOIN moz_places ON moz_places.id = moz_historyvisits.place_id
+        JOIN moz_origins ON moz_origins.id = moz_places.origin_id
+        WHERE moz_origins.host = :host AND visit_date < :before",
+        rusqlite::named_params! {
+            ":host": host,
+            ":before": before,
+        },
+        |r| r.get(0),
+        true,
+    )?;
+    Ok(count)
+}
```

Comment:
it would be great to get a test in this file, copying the same shape as the existing tests at the end of the file.